isos/appliance-xtra.sh

```diff
@@ -0,0 +1,185 @@
+cp ${DIR}/appliance/*.service $(rootfs_dir $PKGDIR)/etc/systemd/system/
+cp ${DIR}/appliance/*-setup $(rootfs_dir $PKGDIR)/etc/systemd/scripts
+
+mkdir -p $(rootfs_dir $PKGDIR)/etc/systemd/system/vic.target.wants
+ln -s /etc/systemd/system/vic-init.service $(rootfs_dir $PKGDIR)/etc/systemd/system/vic.target.wants/
+ln -s /etc/systemd/system/nat.service $(rootfs_dir $PKGDIR)/etc/systemd/system/vic.target.wants/
+ln -s /etc/systemd/system/permissions.service $(rootfs_dir $PKGDIR)/etc/systemd/system/vic.target.wants/
+ln -s /lib/systemd/system/multi-user.target $(rootfs_dir $PKGDIR)/etc/systemd/system/vic.target.wants/
+
+# disable networkd given we manage the link state directly
+rm -f $(rootfs_dir $PKGDIR)/etc/systemd/system/multi-user.target.wants/systemd-networkd.service
+rm -f $(rootfs_dir $PKGDIR)/etc/systemd/system/sockets.target.wants/systemd-networkd.socket
+
+# Disable time synching.  We'll use toolbox for this.
+rm -f $(rootfs_dir $PKGDIR)/etc/systemd/system/sysinit.target.wants/systemd-timesyncd.service
+
+# change the default systemd target to launch VIC
+ln -sf /etc/systemd/system/vic.target $(rootfs_dir $PKGDIR)/etc/systemd/system/default.target
+
+# do not use the systemd dhcp client
+rm -f $(rootfs_dir $PKGDIR)/etc/systemd/network/*
+cp ${DIR}/base/no-dhcp.network $(rootfs_dir $PKGDIR)/etc/systemd/network/
+
+# do not use the default iptables rules - nat-setup supplants this
+rm -f $(rootfs_dir $PKGDIR)/etc/systemd/network/*
+
+#
+# Set up component users
+#
+
+chroot $(rootfs_dir $PKGDIR) groupadd -g 1000 vicadmin
+chroot $(rootfs_dir $PKGDIR) useradd -u 1000 -g 1000 -G systemd-journal -m -d /home/vicadmin -s /bin/false vicadmin
+
+# Group vic should be used to run all VIC related services.
+chroot $(rootfs_dir $PKGDIR) groupadd -g 1001 vic
+chroot $(rootfs_dir $PKGDIR) usermod -a -G vic vicadmin
+
+cp -R ${DIR}/vicadmin/* $(rootfs_dir $PKGDIR)/home/vicadmin
+chown -R 1000:1000 $(rootfs_dir $PKGDIR)/home/vicadmin
+
+# so vicadmin can read the system journal via journalctl
+install -m 755 -d $(rootfs_dir $PKGDIR)/etc/tmpfiles.d
+echo "m  /var/log/journal/%m/system.journal 2755 root systemd-journal - -" > $(rootfs_dir $PKGDIR)/etc/tmpfiles.d/systemd.conf
+
+chroot $(rootfs_dir $PKGDIR) mkdir -p /var/run/lock
+chroot $(rootfs_dir $PKGDIR) chmod 1777 /var/run/lock
+chroot $(rootfs_dir $PKGDIR) touch /var/run/lock/logrotate_run.lock
+chroot $(rootfs_dir $PKGDIR) chown root:vic /var/run/lock/logrotate_run.lock
+chroot $(rootfs_dir $PKGDIR) chmod 0660 /var/run/lock/logrotate_run.lock
+
+## main VIC components
+# tether based init
+cp ${BIN}/vic-init $(rootfs_dir $PKGDIR)/sbin/vic-init
+
+cp ${BIN}/{docker-engine-server,port-layer-server,vicadmin} $(rootfs_dir $PKGDIR)/sbin/
+cp ${BIN}/unpack $(rootfs_dir $PKGDIR)/bin/
+
+if [ -n "${APPLIANCE_OUTNAME}" ]; then
+    APPLIANCE_NAME=$(basename ${APPLIANCE_OUTNAME})
+else
+    APPLIANCE_NAME=appliance-xtra.iso
+fi
+
+echo Appliance name: $APPLIANCE_NAME
+
+if [ -n "${XTRABIN}" ]; then
```

Comment:
We've already checked that this is set.

(However, one way to deduplicate the logic here with the logic in `appliance.sh` is to leave this check here, remove the check above, change the default name to `appliance.iso`, and rename this whole file to `appliance.sh` so that there's a single script that optionally includes an extra binary.)